src/arrow2/src/compute/cast/binary_to.rs

```diff
@@ -155,6 +155,63 @@ pub fn fixed_size_binary_binary<O: Offset>(
     )
 }
 
+pub fn binary_to_fixed_size_binary<O: Offset>(
+    from: &BinaryArray<O>,
+    size: usize,
+) -> Result<Box<dyn Array>> {
+    if let Some(validity) = from.validity() {
+        // Ensure all valid elements have the right size
+        for (value, valid) in from.values_iter().zip(validity) {
+            if valid && value.len() != size {
+                return Err(Error::InvalidArgumentError(
+                    format!(
+                        "element has invalid length ({}, expected {})",
+                        value.len(),
+                        size
+                    )
+                    .to_string(),
+                ));
+            }
+        }
+
+        // Copy values to new buffer, accounting for validity
+        let mut values: Vec<u8> = Vec::new();
+        let offsets = from.offsets().buffer().iter();
+        for (off, valid) in offsets.zip(validity) {
+            if valid {
+                values.extend(from.values().clone().sliced(off.to_usize(), size).iter());
+            } else {
+                values.extend(std::iter::repeat(0u8).take(size));
+            }
+        }
```

Comment:
```
let from_values = from.values();
...
let start = off.to_usize();
let end = start + size;
values.extend(&from_values[start..end]);
```

Instead of cloning the values in every iteration, you could take a slice directly from the reference